scripts/report_scripts/variantreport_p_sample.Rmd

```diff
@@ -406,19 +406,62 @@ msig_stable_complete <- bind_cols(muts_all.df,msig_all,bulk_all.df)
 ```
 
 
+
 ```{r collapse_identicals, message=F, warning=F, include = FALSE, eval = executeDeconvolution}
 # When multiple columns look like the same, the deconvolution will not work, because the function can't distinguish 
 # between those columns. The workaround for now is to identify those equal columns and merge them into one, returning also
 # a vector with the information about which of the columns were merged. 
 
-msig_stable_unique <- dedupe(msig_stable)
+# deduplicate dataframe
+# fixme: This can be simplified and it also does reduntant repetition of loops
+
+msig_stable_transposed <- t (msig_stable)
+msig_stable_transposed <- as.data.frame(cbind(variants = colnames(msig_stable), msig_stable_transposed))
+
+dupes_variants <- duplicated (msig_stable_transposed[,-1], fromLast=TRUE)
+allDupes_variants <- dupes_variants | duplicated (msig_stable_transposed[,-1], fromLast=FALSE)
+
+# fixme isn't that a case for "distinct" ?
+if (any (allDupes_variants)) {
+   var_deduped_transposed <- msig_stable_transposed[!dupes_variants,]
+}
+var_dedup <- as.data.frame(t(var_deduped_transposed))
```

Comment:
This is not good.  Theres' a case where var_deduped_transposed is not defined. 

Generally, avoid top level conditional expressions in assignments.  Move them to the right hand side of the assignment.